org.lflang/src/org/lflang/generator/ts/sourcemap/SourceMapSegment.kt

```diff
@@ -0,0 +1,205 @@
+package org.lflang.generator.ts.sourcemap
+
+import kotlinx.serialization.*
+
+/**
+ * Represents a source map segment that follows the specification provided in
```

Comment:
What is "Source Map Revision 3 Proposal"? Is this a feature for mapping source code locations in target code back to locations in LF code? If so, is this applicable also to other languages?

Reply:
1. It is [the standard](https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/edit#heading=h.djovrt4kdvga) for how to map between TS, JS, and uglified/compressed JS. I should clarify that comment.
2. Yes, that was the intention when I created these classes -- mapping from TS back to LF.
3. It might be applicable? Perhaps that should be discussed. We could use one way to establish mappings for C and C++ (line directives), another way for TS (source maps), and another way for Python and Rust. However, I have heard that we wish to report where errors come from in generated code in addition to knowing where they originate in the source, so maybe the special functionality from line directives (line numbers being adjusted by the compiler when it reports errors) isn't really useful to us. It also isn't clear yet how source maps will hook into existing debugging tools for TS, so it is possible that their special functionality isn't useful to us either. It therefore seems possible that we should choose a single way for establishing these mappings that can be reused across target languages, whether that is source maps, something like line directives, or something of our own design.

Reply:
For C++ I have refrained from using the line directives. For one part this is due to aesthetic aspects, but for the other part it is also because I find that errors reported from the C++ compiler can easily become misleading. Especially, in larger projects where we might have several additional libraries written in target code. To debug certain compiler errors, I think its important to be able to investigate the generated source code (even as an end-user).

Reply:
Yes, ideally we'd be reporting the error in the right location of the LF file while reporting also to which file and line number this corresponds in the generated code.